ethapi/api.go

```diff
@@ -1806,13 +1847,7 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 		"logs":              receipt.Logs,
 		"logsBloom":         &receipt.Bloom,
 		"type":              hexutil.Uint(tx.Type()),
-	}
-	// Assign the effective gas price paid
-	if header.BaseFee == nil {
-		fields["effectiveGasPrice"] = hexutil.Uint64(tx.GasPrice().Uint64())
-	} else {
-		gasPrice := new(big.Int).Add(header.BaseFee, tx.EffectiveGasTipValue(header.BaseFee))
-		fields["effectiveGasPrice"] = hexutil.Uint64(gasPrice.Uint64())
```

Comment:
This should be how we calculate `effectiveGasPrice` = `baseFee` + `effectiveGasTip` for dynamic fee txs

Reply:
First statement is for pre-1559, second is for 1559 and on ward

Reply:
I.e you deleted this and used pre-1559 rule for all txs

Reply:
Logic actually moved up one block, not removed or wrong